applications/crossbar/src/crossbar_maintenance.erl

```diff
@@ -867,51 +888,74 @@ create_app(AppPath, MetaData, MasterAccountDb) ->
             io:format(" saved app ~s as doc ~s~n", [wh_json:get_value(<<"name">>, JObj)
                                                     ,wh_doc:id(JObj)
                                                    ]),
-            maybe_add_icons(AppPath, wh_doc:id(JObj), MasterAccountDb);
+            maybe_add_images(AppPath, wh_doc:id(JObj), MetaData, MasterAccountDb);
         {'error', _E} ->
             io:format(" failed to save app ~s to ~s: ~p~n", [wh_json:get_value(<<"name">>, MetaData), MasterAccountDb, _E])
     end.
 
--spec maybe_add_icons(file:filename(), ne_binary(), ne_binary()) -> 'ok'.
-maybe_add_icons(AppPath, AppId, MasterAccountDb) ->
-    try find_icons(AppPath) of
-        {'ok', Icons} -> add_icons(AppId, MasterAccountDb, Icons)
+-spec delete_old_images(ne_binary(), wh_json:object(), ne_binary()) -> 'ok'.
+delete_old_images(AppId, MetaData, MasterAccountDb) ->
+    Icons       = [wh_json:get_value(<<"icon">>, MetaData)],
+    Screenshots = wh_json:get_value(<<"screenshots">>, MetaData, []),
+
+    _ = [safe_delete_image(MasterAccountDb, AppId, X) || X <- Icons],
+    _ = [safe_delete_image(MasterAccountDb, AppId, X) || X <- Screenshots],
+    'ok'.
+
+-spec safe_delete_image(ne_binary(), ne_binary(), ne_binary()) -> 'ok'.
```

Comment:
Spec should be `api_binary()` for third arg, since it can be `'undefined' | ne_binary()`. See [wh_types.hrl](https://github.com/2600hz/kazoo/blob/master/core/whistle-1.0.0/include/wh_types.hrl#L64) for a lot of the available types.